src/types/k8s.go

```diff
@@ -23,18 +49,29 @@ type ZarfState struct {
 // DeployedPackage contains information about a Zarf Package that has been deployed to a cluster
 // This object is saved as the data of a k8s secret within the 'Zarf' namespace (not as part of the ZarfState secret).
 type DeployedPackage struct {
-	Name       string      `json:"name"`
-	Data       ZarfPackage `json:"data"`
-	CLIVersion string      `json:"cliVersion"`
-
-	DeployedComponents []DeployedComponent `json:"deployedComponents"`
-	ConnectStrings     ConnectStrings      `json:"connectStrings,omitempty"`
+	Name               string                        `json:"name"`
+	Data               ZarfPackage                   `json:"data"`
+	CLIVersion         string                        `json:"cliVersion"`
+	Generation         int                           `json:"generation"`
+	DeployedComponents []DeployedComponent           `json:"deployedComponents"`
+	ComponentWebhooks  map[string]map[string]Webhook `json:"componentWebhooks,omitempty"`
+	ConnectStrings     ConnectStrings                `json:"connectStrings,omitempty"`
 }
 
 // DeployedComponent contains information about a Zarf Package Component that has been deployed to a cluster.
 type DeployedComponent struct {
-	Name            string           `json:"name"`
-	InstalledCharts []InstalledChart `json:"installedCharts"`
+	Name               string           `json:"name"`
+	InstalledCharts    []InstalledChart `json:"installedCharts"`
+	Status             ComponentStatus  `json:"status"`
+	ObservedGeneration int              `json:"observedGeneration"`
+}
+
+// Webhook contains information about a Component Webhook operating on a Zarf package secret.
+type Webhook struct {
+	Name                string `json:"name"`
+	WaitDurationSeconds int    `json:"waitDurationSeconds,omitempty"`
+	Status              string `json:"status"`
```

Comment:
isnt this not a `string`, but a `WebhookStatus`?

Reply:
Good catch, updated to use type `WebhookStatus`